linera-views/src/collection_view.rs

```diff
@@ -305,7 +306,7 @@ where
     /// ```
     pub fn remove_entry(&mut self, short_key: Vec<u8>) {
         *self.hash.get_mut() = None;
-        if self.was_cleared {
+        if self.delete_storage_first {
             self.updates.get_mut().remove(&short_key);
```

Comment:
This optimization might deserve a comment. For instance:
```
// Optimization: No need to mark `short_key` for deletion as we are going to remove all the keys at once.
```

Reply:
Ok.